File: DesafioAndroidConcrete/app/src/main/java/com/marcos/desafioandroidconcrete/viewmodel/PullRequestViewModel.java
package com.marcos.desafioandroidconcrete.viewmodel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.marcos.desafioandroidconcrete.domain.PullRequest;
import com.marcos.desafioandroidconcrete.request.RepositoryRequest;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

/**
 * Created by marco on 19,Abril,2020
 */
public class PullRequestViewModel extends ViewModel {
    public MutableLiveData<List<PullRequest>> pullRequestMutableLiveData = new MutableLiveData<>();

    public List<PullRequest> getPullRequestCache() {
        return pullRequestMutableLiveData.getValue();
    }

    public void getPullRequest(String token, String ownerName, String repositoryName) {
        RepositoryRequest.getInstance().fetchPullRequest(token, ownerName, repositoryName).enqueue(new Callback<List<PullRequest>>() {
            @Override
            public void onResponse(Call<List<PullRequest>> call, Response<List<PullRequest>> response) {
                if (response.isSuccessful()) {
                    pullRequestMutableLiveData.setValue(response.body());
                } else {
                    pullRequestMutableLiveData.setValue(null);
                }
            }

            @Override
            public void onFailure(Call<List<PullRequest>> call, Throwable t) {
                pullRequestMutableLiveData.setValue(null);
            }
        });
    }

}

File: app/src/main/java/com/joaodomingos/desafio_android/ui/view_models/PullRequestViewModel.kt
package com.joaodomingos.desafio_android.ui.view_models

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Transformations
import androidx.lifecycle.ViewModel
import androidx.paging.LivePagedListBuilder
import androidx.paging.PagedList
import com.joaodomingos.desafio_android.api.NetworkService
import com.joaodomingos.desafio_android.api.State
import com.joaodomingos.desafio_android.models.PullRequestModel
//import com.joaodomingos.desafio_android.ui.data_source.PullRequestDataSource
//import com.joaodomingos.desafio_android.ui.factory.PullRequestDataSourceFactory
import io.reactivex.disposables.CompositeDisposable
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class PullRequestViewModel: ViewModel() {
    val TAG = this.javaClass
    val pullRequestMutableLiveData: MutableLiveData<List<PullRequestModel>> by lazy {
        MutableLiveData<List<PullRequestModel>>()
    }

    var pullRequestsList: ArrayList<PullRequestModel> = ArrayList()

    fun loadPullRequests(creator: String, repository: String) {
        State.LOADING

        NetworkService.getService().getPullRequestListItens(creator, repository)
            .enqueue(object : Callback<List<PullRequestModel>> {
                override fun onFailure(call: Call<List<PullRequestModel>>?, t: Throwable) {
                    State.ERROR
                }

                override fun onResponse(
                    call: Call<List<PullRequestModel>>?,
                    response: Response<List<PullRequestModel>>?
                ) {
                    State.DONE

                    if (response != null) {
                        if (!response.isSuccessful) {
                            State.ERROR
                            return
                        }

                        response.body()?.let { pullRequestsList.addAll(it) }
                        pullRequestMutableLiveData.postValue(pullRequestsList)
                    } else {
                        State.ERROR
                    }
                }

            })
    }
}